definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js

```diff
@@ -0,0 +1,803 @@
+
+declare interface Cypress$Core {
+  /**
+   *
+   * @see https://docs.cypress.io/api/cypress-api/custom-commands.html
+   */
+  addChildCommand(name: string, fn: Function): void,
+  addDualCommand(name: string, fn: Function): void,
+  addParentCommand(name: string, fn: Function): void,
+
+  /**
+   *
+   * @see https://docs.cypress.io/api/cypress-api/config.html
+   */
+  config(): { ... },
+  config(key: string): any,
+  config(key: string, value: any): void,
+  config(object: { ... }): void,
+
+  /**
+   * @see https://docs.cypress.io/api/cypress-api/env.html
+   */
+  env(): Cypress$Env,
+  env(name: string): string,
+  env(name: string, value: string): void,
+  env(object: Cypress$Env, ...rest: Array<void>): void,
+
+  /**
+   * @see https://docs.cypress.io/api/utilities/_.html
+   */
+  _: { [key: string]: () => any, ... },
+  /**
+   * @see https://docs.cypress.io/api/utilities/$.html
+   */
+  $: {
+    (): any;
+    [key: string]: () => any,
+    ...
+  },
+  /**
+   * @see https://docs.cypress.io/api/utilities/blob.html
+   */
+  Blob: () => any,
+  /**
+   * @see https://docs.cypress.io/api/utilities/minimatch.html
+   */
+  minimatch: () => any,
+  /**
+   * @see https://docs.cypress.io/api/utilities/moment.html
+   */
+  moment: () => any,
+  /**
+   * @see https://docs.cypress.io/api/utilities/promise.html
+   */
+  Promise: ((resolve: Function, reject: Function) => void) => Promise<any>,
+  /**
+   * @see https://docs.cypress.io/api/utilities/sinon.html
+   */
+  sinon: { [key: string]: any, ... },
+
+  /**
+   * @see https://docs.cypress.io/api/cypress-api/cookies.html
+   */
+  Cookies: {|
+    debug(enabled: boolean, options?: Cypress$DebugOptions): void,
+    preserveOnce(...names: Array<string>): void,
+    defaults(options: {|
+      preserve: string | Array<string> | RegExp | (cookie: any) => boolean,
+    |}): void
+  |},
+
+  /**
+   * @see https://docs.cypress.io/api/cypress-api/custom-commands.html
+   */
+  Commands: {|
+    add: (
+      name: string,
+      options: {|
+        prevSubject: boolean | string | Array<any>,
+      |} | Function,
+      callbackFn?: Function,
+    ) => void,
+    overwrite: (name: string, callbackFn: Function) => void,
+  |},
+
+  /**
+   *
+   * @see https://docs.cypress.io/api/cypress-api/dom.html
+   */
+  dom: {|
+    isAttached: (element: any) => boolean,
+    isDescendent: (parent: any, element: any) => boolean,
+    isDetached: (element: any) => boolean,
+    isDocument: (node: any) => boolean,
+    isDom: (element: any) => boolean,
+    isElement: (element: any) => boolean,
+    isFocusable: (element: any) => boolean,
+    isFocused: (element: any) => boolean,
+    isHidden: (element: any) => boolean,
+    isJquery: (element: any) => boolean,
+    isScrollable: (element: any) => boolean,
+    isVisible: (element: any) => boolean,
+    isWindow: (element: any) => boolean,
+    unwrap: (element: any) => Array<any>,
+    wrap: (element: any) => any,
```

Comment:
**suggestion:** can't all those `any` be replaced with something like `Element` or `HTMLElement`?

Reply:
Based on how this usage is documented,
```
cy.get('img').then(($el) => {
  Cypress.dom.isVisible($el) // true
})
```

I tried the following in my code
```
cy.get(selectors.loginLink).then((ele) => {
  console.log(ele);
});
```
Which gave me
<img width="531" alt="Screen Shot 2020-09-25 at 8 32 56 am" src="https://user-images.githubusercontent.com/12436524/94206927-ea5d6800-ff09-11ea-9407-1613c432d855.png">

From what I can tell cypress has their own wrapped objects for all these things. Even in their docs when it says it should return a dom element, they actually mean it's an instance of `Cypress$Global` which annoyed me so much when typing it